gradle/java.gradle

```diff
@@ -218,28 +193,15 @@ def isJavaVersionAllowed(JavaVersion version) {
   return true
 }
 
-/**
- * For a given java home return the location of java executable
- */
-static String toExecutable(String javaHome) {
-  return Objects.requireNonNull(javaHome) + "/bin/java"
-}
-
-/**
- * Returns java home for a given version or {@code null} if not found
- */
-String findJavaHome(JavaVersion version) {
-  def javaHome = System.getenv("JAVA_${version.majorVersion}_HOME")
-  if (javaHome == null) {
-    return null
+// We default to testing with Java 11 for most tests, but some tests don't support it, where we change
+// the default test task's version so commands like `./gradlew check` can test all projects regardless
+// of Java version.
```

Comment:
nice

Reply:
@anuraaga Have you tested that this interplays correctly with test rule below? I think that when java version is requested explicitly by "testJava15", then we should not silently decrease it to satisfy "maxJavaVersion".

Reply:
Yeah that's the main reason I added it here, above the test rule. Though I don't think we have any situation to trigger ambiguity either because we have no case where minJavaVersionForTests is higher than 11. Even if we added it, it should work though.